AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java

```diff
@@ -0,0 +1,200 @@
+import com.ooyala.android.configuration.Options;
+import com.ooyala.sample.R;
+import com.ooyala.android.ui.AbstractOoyalaPlayerLayoutController.DefaultControlStyle;
+import com.ooyala.android.ui.OptimizedOoyalaPlayerLayoutController;
+import com.ooyala.android.util.DebugMode;
+
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Observable;
+import java.util.Observer;
+
+/**
+ * This activity illustrates how you can create, play a basic playback video
+ * and fast switching streams by setEmbedCode() method
+ */
+public class SetEmbedCodePlayerActivity extends Activity implements OnClickListener, Observer, EmbedTokenGenerator, AdvertisingIdUtils.IAdvertisingIdListener {
+    private static final String TAG = "SetEmbedCodeApp";
+    private OoyalaPlayer player;
+
+    private Button setEmbed;
+
+    private final String APIKEY = "";
+    private final String SECRET = "";
+
+    private final String PCODE = "c0cTkxOqALQviQIGAHWY5hP0q9gU";
+    private final String EMBEDCODE_1 = "oza2pxODE6Vm_TTHPKJ68gBD8JQ7oTuj";
+    private final String EMBEDCODE_2 = "E4bDRwZTE6rMB8oYrzOsuHSPz0XM0dAV";
+    private String embedCode = EMBEDCODE_1;
+
+    private final String ACCOUNT_ID = "pbk-373@ooyala.com";
+    private final String PLAYERDOMAIN = "http://www.ooyala.com";
+
+    public static String getName() {
+        return "Set Embed Code player activity";
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        Log.d(TAG, "TEST - onCreate");
+        super.onCreate(savedInstanceState);
+        Thread.setDefaultUncaughtExceptionHandler(onUncaughtException);
+        try {
+            setContentView(R.layout.player_set_embed_code_layout);
+        } catch (Exception e) {
+            DebugMode.logE( TAG, "Caught!", e );
+        }
+
+        setEmbed = (Button) findViewById(R.id.setEmbed);
+        setEmbed.setOnClickListener(this);
+
+        // optional localization
+        // LocalizationSupport.useLocalizedStrings(LocalizationSupport.loadLocalizedStrings("ja_JP"));
+        PlayerDomain domain = null;
+        try {
+            domain = new PlayerDomain(PLAYERDOMAIN);
+        } catch (Exception e) {
+            // TODO Auto-generated catch block
+            DebugMode.logE(TAG, "Caught!", e);
+        }
+
+        final FCCTVRatingConfiguration tvRatingConfiguration = new FCCTVRatingConfiguration.Builder().setDurationSeconds(5).build();
+
+        player = new OoyalaPlayer(PCODE, domain, this, new Options.Builder().setTVRatingConfiguration( tvRatingConfiguration ).setBypassPCodeMatching(true).setUseExoPlayer(true).build());
+        OptimizedOoyalaPlayerLayoutController layoutController = new OptimizedOoyalaPlayerLayoutController(
+                (OoyalaPlayerLayout) findViewById(R.id.player),
+                player,
+                DefaultControlStyle.AUTO
+        );
+        player.setAdsSeekable(true); // this will help us skip ads if need be.
```

Comment:
Isn't this also going to let you manually skip around during an ad?

Reply:
This sample was cloned from SDK ooyalaTestApp.  We don't  need here ads , so i removed all pieces of ads code.